### PR TITLE
Yandex: use ortb2 info & Core: add webdriver flag

### PR DESCRIPTION
**Pull request by chernodub**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

Used the additional user/device information provided by Prebid to be able to increase the value of the ads for the users.

### CODE REVIEW

File: src/types/ortb2.d.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+export namespace Ortb2 {
```
Comment (chernodub) [inline src/types/ortb2.d.ts:1]: I've allowed myself to slightly improve the types for ortb 2 object in bidder request (according to the [docs](https://docs.prebid.org/dev-docs/bidder-adaptor.html#note-on-ortb-adapters-1:~:text=across%20ad%20units.-,ortb2,-is%20the%20global)). If this change goes against your practices, I could send it via an additional PR or just discard. You are welcome to let me know, thanks

File: modules/yandexBidAdapter.js [hunk 2]
```diff
@@ -493,4 +481,11 @@ function addRTT(url, rtt) {
   return url;
 }
 
+/** @returns {{webdriver: boolean}} */
+function getDeviceExt() {
+  return {
+    webdriver: self.navigator?.webdriver === true
```
Comment (patmmccann) [inline modules/yandexBidAdapter.js:487]: please add this to core, not just your adaper. perhaps, device.ext.webdriver

Comment (patmmccann) [reply modules/yandexBidAdapter.js:487]: please only populate if true in core, you're selcome to make it look however you want on your payload

Comment (patmmccann) [reply modules/yandexBidAdapter.js:487]: perhaps https://github.com/prebid/Prebid.js/blob/ff58535df6f60c2289e73ea4b832aef8afed046e/src/fpd/enrichment.js#L112

File: src/types/ortb2.d.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  type Regs = {
+    coppa?: unknown;
+    ext?: { gdpr?: unknown; us_privacy?: unknown };
+  };
+
+  type User = {
+    ext?: unknown;
+  };
+
+  type Ortb2Object = {
```
Comment (patmmccann) [inline src/types/ortb2.d.ts:35]: this isnt correct, ortb object might have site, dooh, or app, but not more than one

File: src/types/ortb2.d.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    w?: number;
+    h?: number;
+    dnt?: 0 | 1;
+    ua?: string;
+    language: string;
+    sua?: {
+      source?: number;
+      platform: unknown;
+      browsers: unknown[];
+      mobile: number;
```
Comment (patmmccann) [inline src/types/ortb2.d.ts:22]: these 3 dont have the '?' ?

File: modules/yandexBidAdapter.js (hunk 2 above)
Comment (chernodub) [reply modules/yandexBidAdapter.js:487]: Will do, thanks for the suggestion!

File: src/types/ortb2.d.ts (hunk 3 above)
Comment (chernodub) [reply src/types/ortb2.d.ts:35]: Sure, I'll add the props `dooh`, `app`. I couldn't find their exact definition based on the usage, so I'll just use `unknown`. If needed, these could definitely be extended in the future.

Comment (patmmccann) [reply src/types/ortb2.d.ts]: i think #5218 kind of covers it

**Review by patmmccann — APPROVED** [review]

File: src/types/ortb2.d.ts (hunk 3 above)
Comment (patmmccann) [reply src/types/ortb2.d.ts]: #5097 actually...

**Pull request merged**